crates/starknet/src/starknet/mod.rs

```diff
@@ -335,6 +335,17 @@ impl Starknet {
     ) -> Result<TransactionHash> {
         add_invoke_transaction::add_invoke_transcation_v1(self, invoke_transaction)
     }
+
+    pub fn get_block_txs_count(&self, block_id: &BlockId) -> Result<u64> {
+        let block;
+        if let Some(block_number) = self.blocks.block_number_from_block_id(block_id) {
+            block = self.blocks.num_to_block.get(&block_number);
+        } else {
+            return Err(Error::NoBlock);
+        }
+
+        Ok(block.unwrap().get_transactions().len() as u64)
```

Comment:
When block_id is of type BlockId::BlockNumber, then it will use the provided number, which will fail on unwrap, because the return type of block is Option and its value will be None. 
Possible solution:
match block {
Some(x) => x.get_transactions().len(),
None => error
}